srtcore/api.cpp

```diff
@@ -824,6 +827,87 @@ int CUDTUnited::connect(const SRTSOCKET u, const sockaddr* name, int namelen, in
    else if (s->m_Status != SRTS_OPENED)
       throw CUDTException(MJ_NOTSUP, MN_ISCONNECTED, 0);
 
+   if (s->m_pUDT->m_bRendezvous)
```

Comment:
I think it should be enough to change this condition into checking if the bound port is equal to the target port, which should be always the case in Rendezvous. This check is also worth doing with caller-listener arrangement